oci/spec_opts_unix.go

```diff
@@ -371,7 +371,7 @@ func WithUserID(uid uint32) SpecOpts {
 			})
 			if err != nil {
 				if os.IsNotExist(err) || err == errNoUsersFound {
-					s.Process.User.UID, s.Process.User.GID = uid, uid
+					s.Process.User.UID, s.Process.User.GID = uid, 0
```

Comment:
Is it really correct to squash any previous non-zero GID here? Might it not be set in the base image or from a previous call to `WithUser` or `WithUIDGID`. The latter two examples are a bit weak, but the base image one I'm less sure of.